src/keychain_mac.cpp

```diff
@@ -101,67 +102,80 @@ void updateError(keychain::Error &err, OSStatus status) {
     }
 }
 
-void handleCFCreateFailure(keychain::Error &err,
-                           const std::string &errorMessage) {
+void setGenericError(keychain::Error &err, const std::string &errorMessage) {
+    err = keychain::Error{};
     err.message = errorMessage;
     err.type = keychain::ErrorType::GenericError;
     err.code = -1;
 }
 
-CFStringRef createCFStringWithCString(const std::string &str,
-                                      keychain::Error &err) {
-    CFStringRef result = CFStringCreateWithCString(
-        kCFAllocatorDefault, str.c_str(), kCFStringEncodingUTF8);
-    if (result == NULL) {
-        handleCFCreateFailure(err, "Failed to create CFString");
+/*! \brief Helper to manage the lifetime of CFObjects */
+template <typename T> struct ScopedCFRef {
+  public:
+    explicit ScopedCFRef(T &&ref) : _ref(std::move(ref)) {}
+    ~ScopedCFRef() {
+        if (_ref)
+            CFRelease(_ref);
     }
+
+    ScopedCFRef(ScopedCFRef &&) noexcept = default;
+    ScopedCFRef(const ScopedCFRef &) = delete;
+    ScopedCFRef &operator=(const ScopedCFRef &) = delete;
+    ScopedCFRef &operator=(ScopedCFRef &&) = delete;
+
+    const T &getCFRef() const { return _ref; }
```

Comment:
Personal opinion on the naming:

```suggestion
    const T get() const { return _ref; }
```

`T` is a pointer (so no need to bother with `&`).